app/services/payments/create_funds.rb

```diff
@@ -0,0 +1,26 @@
+module Payments
+  class CreateFunds
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      payment = payment_builder.create(amount: amount)
```

Comment:
using create here instead of new creates one extra db write, but the bigger problem is that all the db writes then need to be in an atomic transaction